runtime/common/src/account_conversion.rs

```diff
@@ -11,93 +11,85 @@
 // GNU General Public License for more details.
 
 use cfg_primitives::AccountId;
-use cfg_traits::TryConvert;
 use cfg_types::domain_address::{Domain, DomainAddress};
 use pallet_evm::AddressMapping;
 use sp_core::{Get, H160};
 use sp_runtime::traits::Convert;
-use xcm::*;
+use staging_xcm::v3;
+use staging_xcm_executor::traits::ConvertLocation;
 
-/// Common converter code for translating accounts across different
-/// domains and chains.
-pub struct AccountConverter<R, XcmConverter>(core::marker::PhantomData<(R, XcmConverter)>);
-
-impl<R, XcmConverter> AccountConverter<R, XcmConverter> {
```

Comment:
I really do not like splitting this struct into different parts. Account conversion is essential to be stable. Testing this structure is simple, also ensuring it stays the same. 

For different methods and so on it is harder. I would like to roll back all changes related to this one here. Especially if it is just for the sake of having to spare a few generic arguments here and there.

Reply:
Thread here: https://github.com/centrifuge/centrifuge-chain/pull/1756#discussion_r1559647771